source/modulo_new_core/test/python_tests/translators/test_parameters.py

```diff
@@ -0,0 +1,130 @@
+import clproto
+import numpy as np
+import pytest
+import state_representation as sr
+from modulo_new_core.translators.parameter_translators import write_parameter, read_parameter, read_parameter_const
+from rclpy import Parameter
+
+
+def assert_np_array_equal(a: np.array, b: np.array, places=3):
+    try:
+        np.testing.assert_almost_equal(a, b, decimal=places)
+    except AssertionError as e:
+        pytest.fail(f'{e}')
+
+
+def test_parameter_write(parameters):
+    for name, value_types in parameters.items():
+        param = sr.Parameter(name, value_types[0], value_types[1])
+        ros_param = write_parameter(param)
+        assert ros_param.to_parameter_msg() == Parameter(name, value=value_types[0]).to_parameter_msg()
+
+
+def test_parameter_read_write(parameters):
+    for name, value_types in parameters.items():
+        ros_param = Parameter(name, value=value_types[0])
+        param = read_parameter(ros_param)
+        assert ros_param.name == param.get_name()
+        assert param.get_parameter_type() == value_types[1]
+        assert param.get_value() == value_types[0]
+
+        new_ros_param = write_parameter(param)
+        assert ros_param.to_parameter_msg() == new_ros_param.to_parameter_msg()
+
+
+def test_parameter_const_read(parameters):
+    for name, value_types in parameters.items():
+        param = sr.Parameter(name, value_types[0], value_types[1])
+        ros_param = Parameter(name, value=value_types[0])
+
+        new_param = read_parameter_const(ros_param, param)
+        assert new_param.get_name() == param.get_name()
+        assert new_param.get_parameter_type() == param.get_parameter_type()
+        assert new_param.get_value() == param.get_value()
+
+        new_param.set_name("other")
+        assert new_param.get_name() != param.get_name()
+
+
+def test_parameter_non_const_read(parameters):
+    for name, value_types in parameters.items():
+        param = sr.Parameter(name, value_types[0], value_types[1])
+        ros_param = Parameter(name, value=value_types[0])
+
+        param_ref = param
+        new_param = read_parameter(ros_param, param)
+        assert new_param.get_name() == ros_param.name
+        assert new_param.get_parameter_type() == value_types[1]
+        assert new_param.get_value() == value_types[0]
+
+        assert param_ref.get_name() == ros_param.name
+        assert param_ref.get_parameter_type() == value_types[1]
+        assert param_ref.get_value() == value_types[0]
```

Comment:
This is not quite equivalent to the C++ test. You should not construct the `param` with the actual value, otherwise the equality of `new_param == param_ref` is guaranteed, since `new_param == param`. 

I don't know if there is a default constructor in Python, but otherwise maybe you could try to pass `None` as the value, to ensure that `new_param != param` after the read.

```python
param = sr.Parameter(name, None, value_types[1])
```


For reference, in C++ it is:
```c++
// param value is default-constructed
auto param = std::make_shared<state_representation::Parameter<TypeParam>>("test");
// ros param has non-default value
rclcpp::Parameter ros_param("test", value);

// parameter reference copy has default value
auto param_ref = param;

// the parameter read updates the default value to the non-default
read_parameter(ros_param, param);
```